bin/reth/src/utils.rs

```diff
@@ -176,3 +185,86 @@ impl ListFilter {
         self.len = len;
     }
 }
+
+/// Bitfinity Genesis
+pub async fn bitfinity_genesis(url: String) -> Result<Arc<ChainSpec>> {
+    let client = ethereum_json_rpc_client::EthJsonRcpClient::new(ReqwestClient::new(url));
+
+    let chain_id = client.get_chain_id().await.map_err(|e| eyre::eyre!(e))?;
+
+    tracing::info!("Bitfinity chain id: {}", chain_id);
+
+    let genesis_block = client
```

Comment:
This is not going to work if the blockchain does not have block 0 anymore.
The genesis block and genesis balances should be fetched lazily only if the importer needs them